cucumber/api/features/secrets_batch.feature

```diff
@@ -91,3 +91,15 @@ Feature: Batch retrieval of secrets
     { "cucumber:variable:secret1": "v3", "cucumber:variable:secret2": "v5", "cucumber:variable:secret3": "v6" }
     """
 
+  Scenario: Returns the correct result for binary secrets
+    Given I create a binary secret value for resource "cucumber:variable:secret3"
+    And I set annotation "conjur/mime_type" to "application/octet-stream"
+    And I add the secret value "v2" to the resource "cucumber:variable:secret2"
+    When I GET "/secrets?variable_ids=cucumber:variable:secret3,cucumber:variable:secret2"
+    Then the binary data is preserved in JSON
+
+  Scenario: Raises error on binary secret with no annotation
+    Given I create a binary secret value for resource "cucumber:variable:secret3"
+    And I add the secret value "v2" to the resource "cucumber:variable:secret2"
+    When I GET "/secrets?variable_ids=cucumber:variable:secret3,cucumber:variable:secret2"
+    Then the HTTP response status code is 500
```

Comment:
will this happen also in a single secret retrieval? do we have a test for that?

Reply:
We do currently have a [cucumber](https://github.com/cyberark/conjur/blob/master/cucumber/api/features/secrets.feature#L83) for single secret retrieval. There is not issue when only returning one secret because we do not attempt to return JSON. We simply reply with the secret data as the body.